src/js/Controllers/UIController.js

```diff
@@ -262,10 +275,12 @@ class UIController {
                 if (disallowedLayout) {
                     rpc.params.displayLayout = prevDisplayLayout;
                 }
-
+                if (rpc.params.displayLayout === 'DEFAULT') {
+                    rpc.params.displayLayout = getDefaultLayout(setDisplayLayoutApp)
+                }
                 store.dispatch(setTemplateConfiguration(rpc.params.displayLayout, rpc.params.appID, rpc.params.dayColorScheme, rpc.params.nightColorScheme));
```

Comment:
I didn't realize that this event was guaranteed to finish before moving on in the thread (I always assumed this was asynchronous), but after repeated tests it looks like that is the case. The OnSystemCapabilityUpdated message is always sent in the correct situations from what I can tell.

Reply:
That was my initial assumption as well but looking at https://redux.js.org/api/store#dispatchaction it looks like dispatch is always synchronous